tests/widgetresize.js

```diff
@@ -0,0 +1,324 @@
+
+		focusEditor( editor );
+
+		widget = editor.editing.view.document.getRoot().getChild( 0 );
+
+		for ( const stub of Object.values( mouseListenerSpies ) ) {
+			stub.resetHistory();
+		}
+		commitStub.resetHistory();
+
+		// It's crucial to have a precisely defined editor size for this test suite.
+		editor.editing.view.change( writer => {
+			const viewEditableRoot = editor.editing.view.document.getRoot();
+			writer.setAttribute( 'style', 'width: 400px; padding: 0px; overflow: hidden', viewEditableRoot );
+		} );
+	} );
+
+	afterEach( () => {
+		editorElement.remove();
+
+		if ( editor ) {
+			return editor.destroy();
+		}
+	} );
+
+	describe( 'plugin', () => {
+		it( 'is loaded', () => {
+			expect( editor.plugins.get( WidgetResize ) ).to.be.instanceOf( WidgetResize );
+		} );
+	} );
+
+	describe( 'mouse listeners', () => {
+		beforeEach( () => {
+			createResizer();
+		} );
+
+		it( 'doesnt break when called with unexpected element', async () => {
+			const unrelatedElement = document.createElement( 'div' );
+
+			editor.plugins.get( WidgetResize )._mouseDownListener( {}, {
+				target: unrelatedElement
+			} );
+		} );
+
+		it( 'passes new width to the options.onCommit()', async () => {
+			const usedResizer = 'top-right';
+			const domParts = getWidgetDomParts( widget, usedResizer );
+			const initialPointerPosition = getElementCenterPoint( domParts.widget, usedResizer );
+			const finalPointerPosition = Object.assign( {}, initialPointerPosition );
+
+			finalPointerPosition.pageX += 20;
+
+			mouseMock.down( editor, domParts.resizeHandle );
+
+			mouseMock.move( editor, domParts.resizeHandle, finalPointerPosition );
+			mouseMock.up();
+
+			expect( commitStub.callCount ).to.be.equal( 1 );
+			sinon.assert.calledWithExactly( commitStub, '120px' );
+		} );
+	} );
+
+	it( 'are detached when plugin is destroyed', async () => {
+		await editor.destroy();
+		const plugin = editor.plugins.get( WidgetResize );
+		editor = null;
+
+		const event = new Event( 'mousedown', { bubbles: true } );
+		document.body.dispatchEvent( event );
+
+		// Ensure nothing got called.
+		expect( plugin._mouseDownListener.callCount ).to.be.equal( 0 );
+	} );
+
+	it( 'nothing bad happens if activeResizer got unset', async () => {
+		createResizer( {
+			isCentered: () => true
+		} );
+
+		const usedResizer = 'top-right';
+		const domParts = getWidgetDomParts( widget, usedResizer );
+		const initialPointerPosition = getElementCenterPoint( domParts.widget, usedResizer );
+
+		editor.plugins.get( WidgetResize )._getResizerByHandle = sinon.stub().returns( null );
+
+		mouseMock.down( editor, domParts.resizeHandle );
+
+		mouseMock.move( editor, domParts.resizeHandle, initialPointerPosition );
+		mouseMock.up();
+	} );
+
+	describe( 'Integration (pixels)', () => {
+		let resizer;
+
+		beforeEach( async () => {
+			resizer = createResizer();
+		} );
+
+		it( 'properly sets the state for subsequent resizes', async () => {
```

Comment:
What does it mean properly? Either the test description needs to be improved or the test needs to have its scenario written down in a comment because parsing what's happening is very time-consuming.

Reply:
It means that the state's width should be set correctly after each of two subsequent resizes. First it should be enlarged a little and the second enlarge should base on the first and be enlarged a little more.

This test has been changed now, I believe it's now clear from the test code alone.